### PR TITLE
Add SMTP adapter

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

- Add SMTP adapter

## Test Plan

Existing tests

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/utopia-php/messaging/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/Utopia/Messaging/Adapter/Email/SMTP.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    public function getName(): string
+    {
+        return static::NAME;
+    }
+
+    public function getMaxMessagesPerRequest(): int
+    {
+        return 1000;
```
Comment (Meldiron) [inline src/Utopia/Messaging/Adapter/Email/SMTP.php:44]: Shouldn't this be 1? I dont think you can use SMTP to send multiple mails in 1 go. You need to loop it

File: src/Utopia/Messaging/Adapter/Email/SMTP.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        $mail->Port = $this->port;
+        $mail->SMTPAuth = !empty($this->username) && !empty($this->password);
+        $mail->Username = $this->username;
+        $mail->Password = $this->password;
+        $mail->SMTPSecure = $this->smtpSecure;
+        $mail->SMTPAutoTLS = $this->smtpAutoTLS;
+        $mail->CharSet = 'UTF-8';
+        $mail->Subject = $message->getSubject();
+        $mail->Body = $message->getContent();
+        $mail->AltBody = \strip_tags($message->getContent());
```
Comment (Meldiron) [inline src/Utopia/Messaging/Adapter/Email/SMTP.php:66]: This caused trouble to me when there was style tag. I used this in Appwrite instead

https://github.com/appwrite/appwrite/blob/79c324527e71aa66dd54b51221acd8fdd31fba31/src/Appwrite/Platform/Workers/Mails.php#L108-L111

File: src/Utopia/Messaging/Adapter/Email/SMTP.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        $mail->setFrom($message->getFromEmail(), $message->getFromName());
+        $mail->addReplyTo($message->getReplyToEmail(), $message->getReplyToName());
+        $mail->isHTML($message->isHtml());
+
+        foreach ($message->getTo() as $to) {
+            $mail->addAddress($to);
```
Comment (Meldiron) [inline src/Utopia/Messaging/Adapter/Email/SMTP.php:72]: In addAddress we can also set 2nd parameter for their name, if we have it

File: src/Utopia/Messaging/Adapter/Email/SMTP.php (hunk 1 above)
Comment (abnegate) [reply src/Utopia/Messaging/Adapter/Email/SMTP.php:44]: Maybe the name is the not the best here, this is how many `to` addresses can be added, not how many emails

File: src/Utopia/Messaging/Adapter/Email/SMTP.php (hunk 2 above)
Comment (abnegate) [reply src/Utopia/Messaging/Adapter/Email/SMTP.php:66]: Interesting that strip_tags doesn't remove it :thinking: will add that here too

File: src/Utopia/Messaging/Adapter/Email/SMTP.php (hunk 3 above)
Comment (abnegate) [reply src/Utopia/Messaging/Adapter/Email/SMTP.php:72]: We only have addresses for now :+1:

**Review by Meldiron — APPROVED** [review]

**Pull request merged**